packages/metro-memory-fs/src/index.js

```diff
@@ -638,12 +638,12 @@ class MemoryFs {
     filePath: FilePath,
     options?:
       | {
-          autoClose?: ?boolean,
-          encoding?: ?Encoding,
+          autoClose?: boolean,
+          encoding?: Encoding,
           fd?: ?number,
-          flags?: ?string,
-          mode?: ?number,
-          start?: ?number,
+          flags?: string,
+          mode?: number,
+          start?: number,
```

Comment:
Adjusted based on https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options. Replacing `new Buffer` with `Buffer.from` resulted in type error for `opts.start` in https://github.com/facebook/metro/pull/193/files#diff-edc370c71b61b9fe317068a63d935be9R1164